Updating and generating documentation for the terraform registry

chrisghill opened the pull request:
closes: https://github.com/massdriver-cloud/massdriver/issues/1054

Docs generated using terraform tooling

WillBeebe commented:
This is really great. What are your thoughts on this being part of a pre-commit? I think that's how it's set up for our bundles

chrisghill commented:
> This is really great. What are your thoughts on this being part of a pre-commit? I think that's how it's set up for our bundles

Yeah, I considered that. There were 2 places I modified the generated files:
 * I removed the `schema_path`, `specification_path` and `type` fields from the docs so they aren't even a part of the documentation (nobody needs to know about them)
 * I added some sentences to the provider overview.
 
 I think we should circle back to this when we can remove the `specification_path` and `type` fields, which is probably the next update.
